library/std/src/io/error.rs

```diff
@@ -69,6 +69,8 @@ impl fmt::Debug for Error {
 enum Repr {
     Os(i32),
     Simple(ErrorKind),
+    // &str is a fat pointer, but &&str is a thin pointer.
+    SimpleMessage(ErrorKind, &'static &'static str),
```

Comment:
Am I reading it right that this will increase the size of `Repr`? I think this could be avoided if the repr was made into a `&'static (ErrorKind, &'static str)` instead.

Reply:
Ah, never mind, I missed the fact the discriminant and `ErrorKind` both fit into the remaining 64/32/16bits of the 2nd usize.

Reply:
Yup. I also added a `#[test]` for the size of it, to make sure it fits.